packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -937,6 +938,47 @@ export class GarbageCollector implements IGarbageCollector {
 		return this.deletedNodes.has(nodePath);
 	}
 
+	/**
+	 * Get this client's estimation of the reference state of the give node
+	 * If the "confirmed" property is false, the state may be inaccurate in some cases, e.g. if an interactive client
+	 * sees an object as unreferenced on load, some time later it will transition to Inactive even if it is revived in the meantime.
+	 *
+	 * NOTE - This does not check for the existence of the given nodePath - non-existent ones will report "Referenced" state
+	 * @param nodePath - Container-relative path to the node
+	 */
+	public getGCReferenceInfo(nodePath: string): IExperimentalFluidGCInfo {
+		// Callers should know not to query for internal reference info for deleted nodes
+		assert(
+			!this.deletedNodes.has("nodePath"),
+			"Shouldn't be getting internal reference info for a deleted node",
+		);
+
+		// None of the states are guaranteed to stay up to date with the current implementation.
+		const confirmed = false;
+		const confirmedAtTimestampMs = this.getLastSummaryTimestampMs();
+
+		if (this.tombstones.includes(nodePath)) {
+			return { state: "Tombstoned", confirmed, confirmedAtTimestampMs };
```

Comment:
This may be a 180 from my previous comment but we should include the unreferenced timestamp here right. That will help callers to know the unreferenced age of this node.